eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/OfferTypes.scala

```diff
@@ -153,17 +149,17 @@ object Offers {
     val quantityMin: Option[Long] = records.get[QuantityMin].map(_.min)
     val quantityMax: Option[Long] = records.get[QuantityMax].map(_.max)
 
-    val nodeIdXOnly: ByteVector32 = records.get[NodeId].get.xonly
+    val nodeId: PublicKey =
+      records.get[NodeId].map(_.publicKey).getOrElse(records.get[Paths].get.paths.head.blindedNodes.last.blindedPublicKey)
 
     val sendInvoice: Boolean = records.get[SendInvoice].nonEmpty
 
     val refundFor: Option[ByteVector32] = records.get[RefundFor].map(_.refundedPaymentHash)
 
     val signature: Option[ByteVector64] = records.get[Signature].map(_.signature)
 
-    val contact: Seq[OnionMessages.Destination] =
-      records.get[Paths].flatMap(_.paths.headOption).map(OnionMessages.BlindedPath).map(Seq(_))
-        .getOrElse(Seq(PublicKey(2.toByte +: nodeIdXOnly), PublicKey(3.toByte +: nodeIdXOnly)).map(nodeId => OnionMessages.Recipient(nodeId, None, None)))
+    val contact: Either[Seq[BlindedRoute], PublicKey] =
+      records.get[Paths].map(_.paths).map(Left(_)).getOrElse(Right(records.get[NodeId].get.publicKey))
```

Comment:
Shouldn't we prioritize the `NodeId` over a blinded path if it's available? Note that this will be redundant with `NodeId` if we make `NodeId` required in all offers.

Reply:
No, if there is a blinded path, we MUST use it. And if there is also a `node_id`, we should just ignore it.